src/App.js

```diff
@@ -13,25 +21,29 @@ class App extends React.Component {
 
   addTask = (event) => {
     event.preventDefault();
-    this.setState({
-      tasks: [
-        ...this.state.tasks, 
-        {
-          task: this.state.inputText, 
-          id: Date.now(), 
-          completed: false
-        }
-      ]
-    });
-    this.setState({
-      inputText: ''
-    })
+    if (this.state.inputText.trim() !== '') {
+      this.setState({
+        tasks: [
+          {
+            task: this.state.inputText, 
+            id: Date.now(), 
+            completed: false
+          },
+          ...JSON.parse(window.localStorage.getItem('tasks'))
+        ]
+      }, () => { window.localStorage.setItem('tasks', JSON.stringify(this.state.tasks)); return this.forceUpdate() });
+      this.setState({
+        inputText: ''
+      })
+    }
   }
 
   clearTasks = (event) => {
+    event.preventDefault();
+    console.log('buttons working dude');
     this.setState({
-      tasks: this.state.tasks.filter(x => Object.values(x).includes(false))
-    })
+      tasks: JSON.parse(window.localStorage.getItem('tasks')).filter(x => Object.values(x).includes(false))
+    }, () => { window.localStorage.setItem('tasks', JSON.stringify(this.state.tasks)); return this.forceUpdate() })
```

Comment:
It looks like you have this in multiple places; why not make a method for it? For example:
```javascript
clearTasks = event => {
  // ...
  this.setState({
    tasks: JSON.parse(... , 
    this.setTasks();
  // ...
};

setTasks = () => {
  window.localStorage
    .setItem('tasks', JSON.stringify(this.state.tasks));
  return this.forceUpdate();
};
```